airflow/executors/kubernetes_executor.py

```diff
@@ -599,13 +599,14 @@ def sync(self) -> None:
                 try:
                     self.kube_scheduler.run_next(task)
                 except ApiException as e:
-                    if e.reason == "BadRequest":
-                        self.log.error("Request was invalid. Failing task")
+                    if e.status in (400, 422):
+                        self.log.error("Pod creation failed with reason %r. Failing task", e.reason)
```

Comment:
```suggestion
                        self.log.error("Failing Task. Pod creation failed with reason: %r, message: %r", e.reason, json.loads(e.body)['message'],)
```

Does it makes sense to include the message too? (warning, untested)

Reply:
in this case, the exception is already logged within `run_pod_async`, which catches, logs, and reraises.

probably anything that is going to immediately reraise shouldn't bother logging (so that the logging is controlled by whatever is handling the exception).  but in this case, that's what the code does so it is probably better to only log once, because it can be a bit verbose. lmkyt

Reply:
Ah, gotcha, no that makes sense. Thanks.